### ISSUE TITLE
Fetcher function is not called for subsequent refetches

### ISSUE THREAD

**Issue by snnsnn** [solidjs/solid#1741]
I was trying to run some benchmarks against `createResource` and run into this issue where the fetcher function is called once for multiple `refetch` invocations.

```tsx
import { Bench } from 'tinybench';
import { createResource as newCreateResource } from '@snnsnn/resource';
import { createEffect, createResource, createRoot } from 'solid-js';

const bench = new Bench({ time: 100 });

const solidCR = async () => {
  await new Promise<void>((done) => {
    createRoot(() => {
      let x = 0;
      const fn = async () => {
        return x++;
      };
      const [res, { refetch }] = createResource(fn);
      let total = 99;
      for (let index = 0; index < total; index++) {
        refetch();
      }
      createEffect(() => {
        console.log(res());
        if (res() === 99) {
          done();
        }
      });
    });
  });
};

const newCR = async () => {
  await new Promise<void>((done) => {
    createRoot(() => {
      let x = 0;
      const fn = async () => {
        return x++;
      };
      const res = newCreateResource<number, any, number>(fn);
      let total = 99;
      for (let index = 0; index < total; index++) {
        res.refetch();
      }
      createEffect(() => {
        if (res.data === 99) {
          done();
        }
      });
    });
  });
};

bench.add('new', newCR).add('original', solidCR);

await bench.run();

console.table(bench.table());
```
Inside browser benchmark hangs, never shows a result.

If we log the resource value inside an effect, it outputs first `undefined` than `0`. So, all 99 invocation makes a single request.

```
createEffect(() => {
  console.log(res()); // Runs once
  if (res() === 99) {
    done();
  }
});
```

And inside vitest benchmark, it also hangs and never completes:

```
   name            hz     min     max    mean     p75     p99    p995    p999     rme  samples
 ⠦ new       3,826.90  0.2032  5.4733  0.2613  0.2369  1.2051  1.3521  2.1872  ±3.44%     1914  
 ⠦ original    0.0000  0.0000  0.0000  0.0000  0.0000  0.0000  0.0000  0.0000  ±0.00%        0  
```

**Comment by atk** [comment]
Can you tell me your node version and vitest setup? In some combinations, vitest loads the server version even though the client version is requested, thus breaking the reactive system.

**Comment by snnsnn** [comment]
I gets stuck in the browser too.

```
$ node -v
v19.4.0
```

```
 "devDependencies": {
   "vitest": "^0.31.1",
}
```

```
 "scripts": {
    "bench": "vitest bench"
  }
```

**Comment by atk** [comment]
And the vite(st) config?

**Comment by snnsnn** [comment]
I did not use any config file. Here is what I did:

1. Installed vitest.
2. Created index.bench.ts file with the content above.
3. Added a script called bench to scripts in package.json
4. Run: pnpm run bench

**Comment by snnsnn** [comment]
I wrote an alternative implementation, with few additional features like:

* Optimistic updates
* Request cancellation
* Out of order rendering of `Suspense` components
* Asyn error handling via Suspense components
* Signal bound async consistency that is tied to the resource directly, removing the need for Transitions.

 I added you as a collaborator if you would like to check it out. It is  done. Readme is up to date. I am working on a small presentation, once that one is done, I will share it.

**Comment by atk** [comment]
Without a correct vite.config.ts, vitest will pull the server version, which is not fully reactive, so your results will be invalid.

**Comment by snnsnn** [comment]
> Without a correct vite.config.ts, vitest will pull the server version, which is not fully reactive, so your results will be invalid.

Ok, but what about the browser version, see it logs once:

https://playground.solidjs.com/anonymous/5030d1d5-08de-44e8-82e2-3d8c7aa9f01d

**Comment by ryansolid** [comment]
Could it be the deduping that is stopping this? We don't refetch on the same tick more than once. I wonder if each refetch in its own microtask would achieve this.

**Comment by snnsnn** [comment]
> We don't refetch on the same tick more than once. I wonder if each refetch in its own microtask would achieve this.

Exactly, but with `queueMicrotask` we send one more request, `setTimeout` runs all 99 of them. 

> The reason for this is so that when we refetch and navigate and hit the same thing we don't send out multiple requests.

I think the resource API provides enough flexibility, no need to be overprotective.

**Comment by ryansolid** [comment]
We needed this for solutions like navigation + mutation in routing so that we could both navigate to the new sections and refresh the existing without refetching the new sections. The router is no aware of all the resources that you use with it, so this deduping has to be blind. 

It looks like I intended it to only dedupe on the same synchronous task and reset on the microtask. I wonder why it only refetches one more time. You microtasks getting scheduled before it I imagine. But that seems unexpected.

**Comment by ryansolid** [comment]
I'm going to close this as stale/won't fix. This is part by design, part of it some weird side effect that won't get looked into async gets refactored in 2.0. I don't see 1.x ever changing it's current behavior.
